### ISSUE TITLE
Purr Data build on Apple Silicon crashes on start (macOS Ventura 13.2.1 (22D68))

### ISSUE THREAD

**Issue by luismrguimaraes** [agraef/purr-data#22]
I'm trying to build and run Purr Data on macOS Ventura 13.2.1 (22D68). 

The build process seems to complete with success:
<img width="348" alt="image" src="https://user-images.githubusercontent.com/100025288/226323924-626c795b-45ab-4187-9b98-41cda681de69.png">

However, when trying to run it, the app just **crashes on start**.
This is what Purr Data shows before the crash on the latest commit (3e2039e28e00bea0b29b2eddf941f7cb1ee2b9c1):
<img width="1112" alt="image" src="https://user-images.githubusercontent.com/100025288/226330163-4238bfcf-64d9-4c2c-9def-2fe946030c9a.png">

I've also tried building in the latest release commit a2f5df4, but the problem remains:
<img width="1191" alt="image" src="https://user-images.githubusercontent.com/100025288/226331785-bd145501-79da-4085-95d3-139042118b28.png">
Running [the already built .dmg](https://github.com/agraef/purr-data/releases/tag/2.19.2) however, everything looks okay.

If I build with `make incremental`, the problem remains, but building with `make light` the app runs correctly.

**Comment by agraef** [comment]
Let me guess: You probably built for the M1/M2 architecture. I have been working on the M1/M2 port, but it doesn't work yet, and I don't really know why. Might be something in the externals, or the fact that we still use a prebuilt nw.js version for Intel. Therefore the builds distributed here are all for Intel cpus which run on M1/M2 Macs just fine with Rosetta 2.

**Comment by luismrguimaraes** [comment]
Yes, the chip is the M2. Forgot to include that "small" detail :D

**Comment by agraef** [comment]
Changed the title so that it reflects the actual issue. I'm keeping this open so that we can hopefully make a native Apple Silicon port work. Issues blocking that port are, from the top of my head:

- There are several externals which are segfaulting in `make check` when compiled for Apple Silicon, and a couple more that are not covered by `make check` currently. The segfaults are likely related to miscompiled C calls due to the discrepancies between normal and variadic C calls on the latest ARM architectures). The core seems to be fine (a natively compiled "light" build will work, IIRC), so there's hope that we can eventually address all these issues.

- We're currently stuck with nw.js 0.67.0 due to serious regressions in later versions, and that version doesn't have an Apple Silicon build available, so we'd still rely on Rosetta for that part of the program, which is unsatisfactory. Specifically, we won't be able to have a native nw.js until either https://github.com/nwjs/nw.js/issues/7979 or https://github.com/nwjs/nw.js/issues/7981 gets fixed.
